gui/proxy_edit.go

```diff
@@ -2,11 +2,55 @@ package gui
 
 import (
 	"github.com/gotk3/gotk3/gtk"
+	"github.com/twstrike/coyim/i18n"
 	"github.com/twstrike/coyim/net"
 )
 
+// this is an array because it must be sorted
+var proxyTypes = []string{
+	"tor-auto",
+	"socks5",
+}
+
+var proxyTypesNames = map[string]i18n.T{
+	"tor-auto": i18n.T("Automatic Tor"),
+	"socks5":   i18n.T("SOCKS5"),
+}
+
+// findProxyTypeFor returns the index of the proxy type given
+func findProxyTypeFor(s string) int {
+	for ix, px := range proxyTypes {
+		if px == s {
+			return ix
+		}
+	}
+
+	return -1
+}
+
+// getProxyTypeNames will yield all i18n proxy names to the function
+func getProxyTypeNames(f func(string)) {
+	for _, px := range proxyTypes {
+		l := i18n.Local(string(proxyTypesNames[px]))
```

Comment:
i18n.Local localizes ("translates") a translatable string to the current
locale.

i18n.T marks a string literal as a translatable string, so it can be
translated in different locales.

In order to have i18n.Local("Hi") translating "Hi" to "Oi" you must have
"Hi" marked as a translatable string, then translate it to "Oi" in your
locale.

xgettext extracts translatable strings from source code filed to .pot so
you can translate them to different locales.

The problem is when you use

msg := "Hi"
i18n.Local(msg)

xgettext is unable to know which string the msg variable is pointing to,
so it does not find "Hi" to be a translatable string.

With this change, you could use something like

msg := i18n.T("Hi")
i18n.Local(msg)

To mark such strings as translatable.

In short: i18n.T is a "tag" for helping xgettext and i18n.Local is the
runtime function to localizing string.

On 2/2/16 11:04, Tania Silva wrote:
> In gui/proxy_edit.go
> <https://github.com/twstrike/coyim/pull/225#discussion_r51588813>:
> 
>> +
>> +// findProxyTypeFor returns the index of the proxy type given
>> +func findProxyTypeFor(s string) int {
>> +	for ix, px := range proxyTypes {
>> +		if px == s {
>> +			return ix
>> +		}
>> +	}
>> +
>> +	return -1
>> +}
>> +
>> +// getProxyTypeNames will yield all i18n proxy names to the function
>> +func getProxyTypeNames(f func(string)) {
>> +	for _, px := range proxyTypes {
>> +		l := i18n.Local(string(proxyTypesNames[px]))
> 
> So, basically the idea is to use this marker |T| instead of to
> use/create function |Local|?
> If so, you don't need this |i18n.Local| :cat:
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twstrike/coyim/pull/225/files#r51588813>.
> 

-- 
*Reinaldo de Souza Jr* | Software Developer
*Thought*Works Brasil

EF84 6530 67A5 1559 5554 D8B2 954A 6BEF AF74 ACD7